src/storages/memory_storage/alter_table.rs

```diff
@@ -3,50 +3,176 @@
+                item.schema.column_defs.remove(column_index);
+
+                item.rows.iter_mut().for_each(|(_, row)| {
+                    if row.0.len() > column_index {
+                        row.0.remove(column_index);
+                    }
+                });
+            }
+            None if if_exists => {}
+            None => {
+                return Err(AlterTableError::DroppingColumnNotFound(column_name.to_owned()).into())
+            }
+        };
+
+        Ok(())
+    }
+}
+
 #[async_trait(?Send)]
 impl AlterTable for MemoryStorage {
-    async fn rename_schema(self, _table_name: &str, _new_table_name: &str) -> MutResult<Self, ()> {
-        Err((
-            self,
-            Error::StorageMsg("[MemoryStorage] alter-table is not supported".to_owned()),
-        ))
+    async fn rename_schema(self, table_name: &str, new_table_name: &str) -> MutResult<Self, ()> {
+        let mut storage = self;
+
+        match storage
+            .items
+            .remove(table_name)
+            .ok_or_else(|| AlterTableError::TableNotFound(table_name.to_owned()))
+        {
+            Ok(mut item) => {
+                item.schema.table_name = new_table_name.to_owned();
+                storage.items.insert(new_table_name.to_owned(), item);
+
+                Ok((storage, ()))
+            }
+            Err(err) => Err((storage, err.into())),
+        }
     }
 
     async fn rename_column(
         self,
-        _table_name: &str,
-        _old_column_name: &str,
-        _new_column_name: &str,
+        table_name: &str,
+        old_column_name: &str,
+        new_column_name: &str,
     ) -> MutResult<Self, ()> {
-        Err((
-            self,
-            Error::StorageMsg("[MemoryStorage] alter-table is not supported".to_owned()),
-        ))
+        let mut storage = self;
+
+        result_into(
+            storage.rename_column_sync(table_name, old_column_name, new_column_name),
+            storage,
+        )
     }
 
-    async fn add_column(self, _table_name: &str, _column_def: &ColumnDef) -> MutResult<Self, ()> {
-        Err((
-            self,
-            Error::StorageMsg("[MemoryStorage] alter-table is not supported".to_owned()),
-        ))
+    async fn add_column(self, table_name: &str, column_def: &ColumnDef) -> MutResult<Self, ()> {
+        let mut storage = self;
+
+        result_into(storage.add_column_sync(table_name, column_def), storage)
     }
 
     async fn drop_column(
         self,
-        _table_name: &str,
-        _column_name: &str,
-        _if_exists: bool,
+        table_name: &str,
+        column_name: &str,
+        if_exists: bool,
     ) -> MutResult<Self, ()> {
-        Err((
-            self,
-            Error::StorageMsg("[MemoryStorage] alter-table is not supported".to_owned()),
-        ))
+        let mut storage = self;
+
+        result_into(
+            storage.drop_column_sync(table_name, column_name, if_exists),
+            storage,
+        )
```

Comment:
Oh, there already exists a solution for this case!

```rust
use crate::result::TrySelf;
```
Then, 
```rust
storage                                                   
    .drop_column_sync(table_name, column_name, if_exists) 
    .try_self(storage)                                    
```


ref. https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv
Maybe it would be better to use name of `into_mut` rather than `try_self`... hmm..
In anyway, we can use `TrySelf` in here for now.

Reply:
There was something good about `result.rs`.
I'll also think about the naming.